project/minchuncho/README.rst

```diff
@@ -0,0 +1,73 @@
+**SewUP** is a tool for sorting out the competition in a Stackelberg game in certain cases.
+
+Problem to Solve
+=================
+A Stackelberg model is a strategic game which consists of a leader and a group of followers, who compete over a quantity while maximizing their profits or minimizing their costs. In the model, the leader acts first and the followers respond afterwards, which goes on and on until everyone settles.  
+However, whether or not the game will converge is up in the air. Gladly, such games take advantages of **backward induction** where the leader predicts followers’ best responses under a given strategy of his. If luckily, those best responses can be rewritten as functions of the leader’s strategy, then the leader can substitute those best responses back to his utility (or cost) function and find the extremum. Obviously, the game will stop with **just one back-and-forth**, which is quite efficient. Note that this approach favors the leader, which is called “first mover’s advantage.”
+
+.. Since the whole process is determined by the leader, it is like the leader is trying to bring the situation to his/her favorable conclusion, hence the project is named “SewUP.”
+
+Prospective Users
+==================
+For those researchers who study resource allcation and incentive mechanisms, especially one-to-many Stackelberg game with certain formats of utility functions, their problems can be examined or even solved by the tool.
+
+.. As a matter of fact, if someone is able to model his competition into a Stackelberg game and hopes to solve it by backward induction, then very likely the utility functions have certain formats.
+
+System Architecture
+====================
+
+.. image:: images/system_architecture.png
+
+Limitations
+-----------
+1. The solution works on the premise that the order of every term in followers' utility functions is at most 2. For example, ``x_1*x_2`` or ``x_3^2`` is allowed but ``x_1*x_2*x_5`` and ``x_2/x_8``\￼ are forbidden.
+2. The number of followers is limited to 100.
+
+API Description
+=================
+.. code-block:: python
+   :linenos:
+
+   import sewup
+   f = sewup.follwer("concave", ["x_1*x_2*y+x_2*x_3", "x_2^2+x_3^2*y", "x_1^2*y"])
+   if f.prop == "concave":
+      x = f.solveEqs() # x written as functions of y, i.e., x(y)
+      l = sewup.leader("concave", "y(x_1+x_2+x_3)")
+      l.substitute(x)
+      y = l.getExtremum()
+      f.substitute(y)
+      print("Nash equalibrium is where x=({0}) and y={1}", f.ans(), l.ans())
+   else:
+      print("Invalid equations.\n")
+
+
+Engineering Infrastructure
+===========================
+1. Automatic build system and how to build your program: CMake
+2. Version control (show how you will use it): Git
+3. Testing framework: Pytest
+4. Documentation: (Will be included in my github repository soon.)
+
+Schedule
+=========
+* Planning phase (6 weeks including 2/21, 2/28, 3/7, 3/14, 3/21, 3/28):
+* Week 1 (4/4): Design of data structures
```

Comment:
It is a productive way to start with designing data structure.  In my experience, the data structure may need continuous adjustment.  We should be fine as long as we are not too rigid with the initial design.

Reply:
Got it, I'll bear in mind and constantly check if my data structures go well with the algorithms.